### PR TITLE
Allows publishing to a  Shared Image Gallery with a different subscription id

### PR DESCRIPTION
**Pull request by sylviamoss**
This adds the subscription option to the `shared_image_gallery_destination`.  This is done by updating the client with the destination subscription id. 

Closes https://github.com/hashicorp/packer/issues/8608

### CODE REVIEW

**Review by nywilken — APPROVED** [review]
This looks good. Nicely done. 

I left a comment about using adding an AzureClientConfig structure but I think that should be a separate PR.

**Comment by nywilken** [comment]
I posted a comment on the thread asking the users to test, but I believe this is good to go. Let's way to hear back before merging.

File: builder/azure/arm/azure_client.go [hunk 1]
```diff
@@ -128,8 +128,8 @@ func byConcatDecorators(decorators ...autorest.RespondDecorator) autorest.Respon
 	}
 }
 
-func NewAzureClient(subscriptionID, resourceGroupName, storageAccountName string,
-	cloud *azure.Environment, SharedGalleryTimeout time.Duration, PollingDuration time.Duration,
+func NewAzureClient(subscriptionID, sigSubscriptionID, resourceGroupName, storageAccountName string,
```
[the remark this answers is not in the document]
Comment (sylviamoss) [reply builder/azure/arm/azure_client.go:131]: Yes! I totally agree and I actually thought about that too but this part of the code can be improved in so many ways and I didn't want to make bigger changes in this PR.
Later I think we can open a tech debt PR to organize the code better, not only this part 🤔

**Comment by sylviamoss** [comment]
I was able to test this anyways with our environment. If they take to long to reply, I think it is ok to merge.

**Review by azr — APPROVED** [review]
nice !

File: builder/azure/arm/config.go [hunk 2]
```diff
@@ -118,31 +119,64 @@ type Config struct {
...
+	// "managed_image_resource_group_name": "TargetResourceGroup"
+	// ```
+	// In HCL2
+	// ```hcl
+	// shared_image_gallery {
+	//     subscription = "00000000-0000-0000-0000-00000000000"
+	//	   resource_group = "ResourceGroup"
+	//	   gallery_name = "GalleryName"
+	//	   image_name = "ImageName"
+	//	   image_version = "1.0.0"
```
Comment (azr) [inline builder/azure/arm/config.go:144]: NIT: 
```suggestion
	//     subscription = "00000000-0000-0000-0000-00000000000"
	//     resource_group = "ResourceGroup"
	//     gallery_name = "GalleryName"
	//     image_name = "ImageName"
	//     image_version = "1.0.0"
```

File: builder/azure/arm/config.go [hunk 3]
```diff
@@ -118,31 +119,64 @@ type Config struct {
...
+	// "managed_image_resource_group_name": "TargetResourceGroup"
+	// ```
+	// In HCL2
+	// ```hcl
+	// shared_image_gallery_destination {
+	//     subscription = "00000000-0000-0000-0000-00000000000"
+	//	   resource_group = "ResourceGroup"
+	//	   gallery_name = "GalleryName"
+	//	   image_name = "ImageName"
+	//	   image_version = "1.0.0"
```
Comment (azr) [inline builder/azure/arm/config.go:174]: ah whoops, this one too: 
```suggestion
	//     subscription = "00000000-0000-0000-0000-00000000000"
	//     resource_group = "ResourceGroup"
	//     gallery_name = "GalleryName"
	//     image_name = "ImageName"
	//     image_version = "1.0.0"
```

**Pull request merged**